### PR TITLE
transferlist.py: port to new TreeView class

### PR DESCRIPTION
**Pull request by mathiascode**
Only the search view remains after this, which will be ported after https://github.com/nicotine-plus/nicotine-plus/pull/2464 is merged. We can then start using Gtk.ColumnView in GTK 4.

### CODE REVIEW

File: pynicotine/gtkgui/widgets/treeview.py [hunk 1]
```diff
@@ -529,15 +547,18 @@ def clear(self):
...
 
-        if "online" in icon_name:
+        if icon_name == USER_STATUS_ICON_NAMES[UserStatus.ONLINE]:
             return _("Online")
 
-        return _("Offline")
+        if icon_name == USER_STATUS_ICON_NAMES[UserStatus.OFFLINE]:
+            return _("Offline")
+
+        return icon_name
```
Comment (slook) [inline pynicotine/gtkgui/widgets/treeview.py:561]: There is never a case when we want the Tooltip Text to read the filename of the icon.

If the user status cannot be determined then "Offline" is the most sensible fallback to return.

Comment (mathiascode) [reply pynicotine/gtkgui/widgets/treeview.py]: There was a conflict between the user status and transfer status columns, but I've resolved it in another way.

**Comment by slook** [comment]
The previous session had just reordered columns. The item in question is with status "Queued" at 0%.

_downloads.json:_
```
16 |  
-- | --
0 | "user.name"
1 | "@@abcdef\\user.name\\01. Jan\\12\\Artist - Title - [user.name]\\Artist - Title (Remix) - [user.name].mp3"
2 | ""
3 | "User logged off"
4 | 20078196
5 | null
6 | {}
```

_Shortly after startup, on switch tab..._
```
Nicotine+ Version: 3.3.0.dev5
GTK Version: 3.24.24
Python Version: 3.9.2 (linux)

Type: <class 'KeyError'>
Value: 'user.name'
Traceback:   File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/gtkgui/application.py", line 708, in on_process_thread_events
    events.process_thread_events()
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/events.py", line 259, in process_thread_events
    self.emit(event_name, *args, **kwargs)
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/events.py", line 226, in emit
    function(*args, **kwargs)
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/transfers.py", line 1386, in _upload_denied
    self.get_file(user, filename, transfer=download)
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/transfers.py", line 1706, in get_file
    self.update_download(transfer)
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/transfers.py", line 2152, in update_download
    events.emit("update-download", transfer, update_parent)
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/events.py", line 226, in emit
    function(*args, **kwargs)
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/gtkgui/transferlist.py", line 325, in update_model
    self.update_parent_rows(transfer)
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/gtkgui/transferlist.py", line 339, in update_parent_rows
    user_path_iter, user_path_child_transfers = self.paths[user_path]
```

`user.name` contains only ASCII characters and a `.` (dot).

Same crash persists on every startup.

File: pynicotine/gtkgui/transferlist.py [hunk 2]
```diff
@@ -327,31 +327,27 @@ def update_model(self, transfer=None, update_parent=True, forceupdate=False):
...
-                path = transfer.path if self.type == "download" else transfer.filename.rsplit("\\", 1)[0]
-                user_path = username + path
-
-                user_path_iter = self.paths.get(user_path)
-                user_iter = self.users.get(username)
+                user_path = username + self.get_transfer_path(transfer)
 
-                if user_path_iter:
-                    self.update_parent_row(user_path_iter, user_path, folder=True)
+                user_path_iter, user_path_child_transfers = self.paths[user_path]
```
Comment (slook) [inline pynicotine/gtkgui/transferlist.py:339]: The path can be None for Queued/Incomplete items that are yet to start, leaving only the username without any folder. A KeyError can occur in Group By User grouping mode.

**Comment by mathiascode** [comment]
Can you check if any issues persist after the latest commit?

**Comment by slook** [comment]
That crash does not happen now.

**Pull request merged**